### PR TITLE
Fixed #26089 -- Removed custom user test models from public API.

### PR DESCRIPTION
**Pull request by charettes**

### CODE REVIEW

File: tests/auth_tests/test_mixins.py [hunk 1]
```diff
@@ -235,7 +235,9 @@ class AView(PermissionRequiredMixin, EmptyResponseView):
 
     def test_permissioned_denied_redirect(self):
         class AView(PermissionRequiredMixin, EmptyResponseView):
-            permission_required = ['auth.add_customuser', 'auth.change_customuser', 'non-existent-permission']
+            permission_required = [
+                'auth_tests.add_customuser', 'auth_tests.change_customuser', 'non-existent-permission'
+            ]
```
Comment (charettes) [inline tests/auth_tests/test_mixins.py]: Will add trailing commas once the build finishes.

File: docs/releases/1.10.txt [hunk 2]
```diff
@@ -448,6 +448,13 @@ output::
         }
     }
 
+The ``auth.CustomUser`` and ``auth.ExtensionUser`` test models were removed.
```
Comment (timgraham) [inline docs/releases/1.10.txt]: ``auth.CustomUser`` and ``auth.ExtensionUser`` test models removed

File: docs/releases/1.10.txt [hunk 3]
```diff
@@ -448,6 +448,13 @@ output::
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Since the introduction of contrib application migrations in Django 1.8 the
```
Comment (timgraham) [inline docs/releases/1.10.txt]: migrations for the contrib apps in Django 1.8,

**Comment by timgraham** [comment]
I guess I should have included "test" in the commit message to avoid any notion that support for custom user models is removed. ;-)

**Comment by charettes** [comment]
Haha didn't even notice before you pointed it out.

**Pull request merged**